test/project_types/script/layers/infrastructure/languages/assemblyscript_project_creator_test.rb

```diff
@@ -5,222 +5,62 @@
-        expected == build
-      end
-
-      subject
-    end
-
-    it "should fetch the latest extension point version if the package is not versioned" do
-      context
-        .expects(:capture2e)
-        .with("npm --userconfig ./.npmrc config set @shopify:registry https://registry.npmjs.com")
-        .returns(fake_capture2e_response)
-      context
-        .expects(:capture2e)
-        .with("npm --userconfig ./.npmrc config set engine-strict true")
-        .returns(fake_capture2e_response)
-      context
-        .expects(:capture2e)
-        .with("npm show @shopify/extension-point-as-fake version --json")
+    it "should setup dependencies" do
+      Script::Layers::Infrastructure::Languages::ProjectCreator.any_instance
+        .expects(:setup_dependencies)
+        .with
         .once
-        .returns([JSON.generate("2.0.0"), OpenStruct.new(success?: true)])
-
-      sdk = mock
-      sdk.expects(:sdk_version)
-      sdk.expects(:toolchain_version)
-      sdk.expects(:versioned?).once.returns(false)
-      sdk.expects(:package).twice.returns("@shopify/extension-point-as-fake")
-      extension_point.expects(:sdks).times(5).returns(stub(all: [sdk], assemblyscript: sdk))
 
-      subject
-      version = JSON.parse(File.read("package.json")).dig("devDependencies", "@shopify/extension-point-as-fake")
-      assert_equal "^2.0.0", version
-    end
-
-    it "should set the specified package version when the package is versioned" do
       context
         .expects(:capture2e)
         .with("npm --userconfig ./.npmrc config set @shopify:registry https://registry.npmjs.com")
         .returns(fake_capture2e_response)
       context
         .expects(:capture2e)
-        .then.with("npm --userconfig ./.npmrc config set engine-strict true")
-        .returns(fake_capture2e_response)
-
-      context
-        .expects(:capture2e)
-        .with("npm show @shopify/extension-point-as-fake version --json")
-        .never
-
-      sdk = mock
-      sdk.expects(:sdk_version)
-      sdk.expects(:toolchain_version)
-      sdk.expects(:package).once.returns("@shopify/extension-point-as-fake")
-      sdk.expects(:versioned?).once.returns(true)
-      sdk.expects(:version).once.returns("file:///path")
-      extension_point.expects(:sdks).times(5).returns(stub(all: [sdk], assemblyscript: sdk))
-
-      subject
-      version = JSON.parse(File.read("package.json")).dig("devDependencies", "@shopify/extension-point-as-fake")
-      assert_equal "file:///path", version
-    end
-
-    it "should raise if the latest extension point version can't be fetched" do
-      context
-        .expects(:capture2e)
-        .with("npm --userconfig ./.npmrc config set @shopify:registry https://registry.npmjs.com")
-        .returns(fake_capture2e_response)
-      context
-        .expects(:capture2e)
-        .then.with("npm --userconfig ./.npmrc config set engine-strict true")
+        .with("npm --userconfig ./.npmrc config set engine-strict true")
```

Comment:
I also define these tedious long strings as constants in the class (in this case in `assemblyscript_project_creator`), so it will be better for maintainability.